Sources/MySQL/Error.swift

```diff
@@ -11,7 +11,7 @@ public enum Error: Swift.Error {
     case inputBind(String)
     case outputBind(String)
     case fetchFields(String)
-    case prepare(String)
+    case prepare(String, UInt32)
```

Comment:
The problem is that this needs to be passed up by the `Connection` class when throwing. Otherwise, once the connection is closed by the `defer` method, the `mysql_errno` will return zero.

I agree, indeed all enums in MySQL error should return both the message and the error code, for better error handling at the application and driver levels.